### PR TITLE
test(types): add unit tests for stream_configured.go and fix validation bugs

### LINKED ISSUE #649: feat: Add Unit Tests for stream_configured.go
### **Problem**

The OLake codebase currently has unit tests for:

* `resolver`
* `flattener`
* `catalog`

However, the file `olake/types/stream_configured.go` does not have any test coverage.

This file handles critical stream configuration logic

---

#### **Proposed Solution**

Add comprehensive unit tests for the logic inside `stream_configured.go`.  
This will help:

* Increase overall test coverage
* Improve stream configuration reliability
* Catch bugs earlier during CI
* Make contributions safer

For reference on how existing unit tests are structured, you can refer to **PR #273**.

---

#### **Notes for Contributors**

* Add tests covering most of the scenarios and edge cases which should indicate both pass and fail scenarios.
* Follow the existing test file structure and naming conventions to maintain consistency with other parts.

### PR DESCRIPTION
**Pull request by HarshitPal25**
- **Summary:** Fixes issues  by correcting validation/behaviour in types/stream_configured.go and adding comprehensive unit tests to prevent regressions.

**What changed:**

- **Code:** Updated types/stream_configured.go to fix validation and parsing edge-cases (cursor handling, filter parsing, and destination selection).
- **Tests:** Added types/stream_configured_test.go with unit tests covering filter parsing, cursor behavior, destination selection, and Validate() scenarios.
- **Why:** The changes ensure StreamConfigured behaves correctly when parsed/validated and catches regressions via unit 
```
# from repo root
go test ./...        # run full test suite
go test ./types -run StreamConfigured -v   # run only the new/related tests
```
**How to verify:**
Files touched:
types/stream_configured.go
types/stream_configured_test.go
**Notes:** This PR closes/fixes #649. Please review the validation logic in Validate() and the new test cases for coverage gaps or additional edge-cases to add.

## Documentation

- [ ] Documentation Link: [link to README, olake.io/docs, or olake-docs]
- [x] N/A (bug fix, refactor, or test changes only)

### CODE REVIEW

**Comment by nayanj98** [comment]
@HarshitPal25 Please sign the CLA and rebase your PR to staging

**Comment by nayanj98** [comment]
@HarshitPal25 Just a reminder can you please rebase your PR to staging and also sign the CLA.

**Comment by HarshitPal25** [comment]
done changed the base to staging

**Comment by nayanj98** [comment]
@HarshitPal25 Great! @vikaxsh will be reviewing your work.

**Comment by HarshitPal25** [comment]
thank you

File: go.mod [hunk 1]
```diff
@@ -18,14 +18,15 @@ require (
...
 	github.com/spf13/viper v1.20.1
 	github.com/stretchr/testify v1.11.1
 	github.com/xitongsys/parquet-go v1.6.2
 	github.com/xitongsys/parquet-go-source v0.0.0-20241021075129-b732d2ac9c9b
 	go.mongodb.org/mongo-driver v1.17.3
-    google.golang.org/grpc v1.71.3
-	google.golang.org/protobuf v1.36.6
+	google.golang.org/grpc v1.74.2
+	google.golang.org/protobuf v1.36.8
 	gopkg.in/natefinch/lumberjack.v2 v2.2.1
```
Comment (vikaxsh) [inline go.mod]: what is the need of version upgrade here?

File: types/stream_configured_test.go [hunk 2]
```diff
@@ -1,570 +1,160 @@
 package types
```
Comment (vikaxsh) [inline types/stream_configured_test.go:1]: Please implement these test cases similar to the approach used in PR #273 for consistency.

- Convert them into table-driven tests using testCases structs for better maintainability.
- Replace t.Fatalf with testify/assert to match the style used across other test files.
- Follow the same naming conventions used for test functions in that PR.

File: go.mod (hunk 1 above)
Comment (HarshitPal25) [reply go.mod]: i was surfing and i found new version is relesed with resolved patches so i upgrade the version

Comment (vikaxsh) [reply go.mod]: Please avoid upgrading the version unless it's absolutely necessary, as it may introduce unexpected failures elsewhere.

Comment (HarshitPal25) [reply go.mod]: i have revert back the version as you said

**Comment by HarshitPal25** [comment]
@vikaxsh i have changed the tests with table driven approach and update the grpc

File: go.mod [hunk 3]
```diff
@@ -18,13 +18,14 @@ require (
 	github.com/lib/pq v1.10.9
 	github.com/parquet-go/parquet-go v0.25.0
 	github.com/rs/zerolog v1.34.0
+	github.com/segmentio/kafka-go v0.4.49
```
Comment (vikaxsh) [inline go.mod:23]: please revert all version upgrades

File: types/stream_configured_test.go [hunk 4]
```diff
@@ -1,570 +1,246 @@
...
```
Comment (vikaxsh) [inline types/stream_configured_test.go:1]: Missing test coverage
- 8 functions have no tests: ID(), Self(), Name(), GetStream(), Namespace(), Schema(),SupportedSyncModes(), GetSyncMode()

Missing edge cases in existing tests:
- TestGetFilter: Missing operators (<, <=, !=, =), quoted strings, OR logic, invalid formats. Not validating second condition values when wantConds == 2.
- TestCursor: Missing empty field, multiple colons, whitespace handling.
- TestGetDestinationDatabase: Missing namespace fallback when both destinationDatabase and icebergOverride are empty/nil.
- TestGetDestinationTable: Missing case when destinationTable is provided (non-empty).
- TestValidate: Only error cases tested. Missing valid cases for:
Valid FULLREFRESH/INCREMENTAL/CDC/STRICTCDC sync modes
CDC/STRICTCDC should skip cursor validation
Valid cursor fields (primary + secondary)
Empty/matching primary keys
Valid/invalid filter in metadata
- TestNormalizationEnabled: Missing false case.
Recommendation: Add tests for all functions and expand edge cases to match the coverage pattern in catalog_test.go.

**Comment by vikaxsh** [comment]
hi @HarshitPal25, is there any update ?

**Comment by HarshitPal25** [comment]
@vikaxsh sorry I didn't get time my exams were going on but now I am free i had done mostly i would update it as soon as possible

**Comment by HarshitPal25** [comment]
@vikaxsh please check i have changed and added what you like in the file

**Comment by Akshay-datazip** [comment]
Thank you so much for your contribution! Our team will get back to this shortly we’re currently working on a higher-priority task, so it may take about a week.

In the meantime, feel free to reach out on Slack anytime. If you run into any issues, please post in the contributing-to-OLake channel and tag me I’ll be happy to help

**Comment by vikaxsh** [comment]
hey @HarshitPal25, please resolve the conflict with base branch and revert go.mod changes

**Comment by Akshay-datazip** [comment]
Hey @HarshitPal25 any update will have to de-assign then issue due to inactivity in some days

**Comment by HarshitPal25** [comment]
@vikaxsh i have revert the changes with the go.mod

**Comment by HarshitPal25** [comment]
@vikaxsh and @Akshay-datazip i have revert back the changes of the go.mod file.

**Comment by vikaxsh** [comment]
hi @HarshitPal25 , please check why github workflows are failing

**Comment by HarshitPal25** [comment]
@vikaxsh yes working on them why my checks are failing well thank you also for staying in touch with me so long and coopration!!!

**Comment by vikaxsh** [comment]
@HarshitPal25 also please also attach a video of running a sync and testing your code

**Comment by HarshitPal25** [comment]
@vikaxsh how i want to make video like giving these faliing check that they are resolved?

**Comment by vikaxsh** [comment]
> @vikaxsh how i want to make video like giving these faliing check that they are resolved?

make video of running olake sync and explain about your implementation

**Comment by vikaxsh** [comment]
hi @HarshitPal25 is there any update?

**Comment by HarshitPal25** [comment]
@vikaxsh sorry I am kinda busy in family function I need some time as I can't get enough time to work on it right now I just need some more days

**Comment by vikaxsh** [comment]
hey @HarshitPal25 ,Please resolve all the review comments by next week; otherwise, this PR will be closed.
